src/core/data/ions/particle_initializers/maxwellian_particle_initializer.h

```diff
@@ -197,9 +187,10 @@ void MaxwellianParticleInitializer<ParticleArray, GridLayout>::loadParticles(
             if (basis_ == Basis::Magnetic)
                 particleVelocity = basisTransform(basis, particleVelocity);
 
-            particles.emplace_back(Particle{cellWeight, particleCharge_,
-                                            AMRCellIndex.template toArray<int>(),
-                                            deltas(deltaDistrib, randGen), particleVelocity});
+            particles.emplace_back(
+                Particle{cellWeight, particleCharge_, AMRCellIndex.template toArray<int>(),
+                         core::ConstArrayFrom<dimension>([&] { return deltaDistrib(randGen); }),
```

Comment:
From what I see what the code was before and based on what you said yesterday, I guess what PGI didn't like was the different return types in if constexprs for the lambda that was passed.

Reply:
weirdly the other lambda was ok